crates/client/src/client/schema/chain.rs

```diff
@@ -325,121 +327,240 @@ impl TryFrom<GasCosts> for fuel_core_types::fuel_tx::GasCosts {
-                    pshh: value.pshh.into(),
-                    pshl: value.pshl.into(),
-                    ret: value.ret.into(),
-                    rvrt: value.rvrt.into(),
-                    sb: value.sb.into(),
-                    sll: value.sll.into(),
-                    slli: value.slli.into(),
-                    srl: value.srl.into(),
-                    srli: value.srli.into(),
-                    srw: value.srw.into(),
-                    sub: value.sub.into(),
-                    subi: value.subi.into(),
-                    sw: value.sw.into(),
-                    sww: value.sww.into(),
-                    time: value.time.into(),
-                    tr: value.tr.into(),
-                    tro: value.tro.into(),
-                    wdcm: value.wdcm.into(),
-                    wqcm: value.wqcm.into(),
-                    wdop: value.wdop.into(),
-                    wqop: value.wqop.into(),
-                    wdml: value.wdml.into(),
-                    wqml: value.wqml.into(),
-                    wddv: value.wddv.into(),
-                    wqdv: value.wqdv.into(),
-                    wdmd: value.wdmd.into(),
-                    wqmd: value.wqmd.into(),
-                    wdam: value.wdam.into(),
-                    wqam: value.wqam.into(),
-                    wdmm: value.wdmm.into(),
-                    wqmm: value.wqmm.into(),
-                    xor: value.xor.into(),
-                    xori: value.xori.into(),
-
-                    aloc: value.aloc_dependent_cost.into(),
-                    cfe: value.cfe.into(),
-                    cfei: value.cfei_dependent_cost.into(),
-                    call: value.call.into(),
-                    ccp: value.ccp.into(),
-                    croo: value.croo.into(),
-                    csiz: value.csiz.into(),
-                    k256: value.k256.into(),
-                    ldc: value.ldc.into(),
-                    logd: value.logd.into(),
-                    mcl: value.mcl.into(),
-                    mcli: value.mcli.into(),
-                    mcp: value.mcp.into(),
-                    mcpi: value.mcpi.into(),
-                    meq: value.meq.into(),
-                    retd: value.retd.into(),
-                    s256: value.s256.into(),
-                    scwq: value.scwq.into(),
-                    smo: value.smo.into(),
-                    srwq: value.srwq.into(),
-                    swwq: value.swwq.into(),
-                    contract_root: value.contract_root.into(),
-                    state_root: value.state_root.into(),
-                    vm_initialization: value.vm_initialization.into(),
-                    new_storage_per_byte: value.new_storage_per_byte.into(),
-                };
-                Ok(fuel_core_types::fuel_tx::GasCosts::new(values.into()))
+                // Detect which version of fuel-tx gas costs to use
+                if value.bsiz.is_some() && value.bldd.is_some() {
```

Comment:
It will not work. Even requesting new optional fields returns an error for old `fuel-core`. You can double check it by using locally old `fuel-core`. It would be nice to double check since we upgraded the `async-graphql` to a new version, maybe new version handles it better. But if it still fails, you just need to create V4 GasCosts=)